### PR TITLE
fix: #ARGB to #RGBA conversion

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Fixes an issue where `#ARGB` ->`#RGBA` and `#AARRGGBB` ->`#RRGGBBAA` were converted improperly when setting background color.

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed an issue where `#ARGB` ->`#RGBA` and `#AARRGGBB` ->`#RRGGBBAA` were converted improperly when setting background color.

### CODE REVIEW

**Review by nornagon — APPROVED** [review]
whoops!

File: shell/common/color_util.cc [hunk 1]
```diff
@@ -35,16 +35,13 @@ namespace electron {
...
-  if (IsHexFormat(color_str)) {
-    if (color_str.length() == 5) {
-      // #ARGB => #RGBA
-      std::swap(color_str[1], color_str[4]);
-    } else {
-      // #AARRGGBB => #RRGGBBAA
-      std::swap(color_str[1], color_str[7]);
-      std::swap(color_str[2], color_str[8]);
-    }
+  std::string color_str;
```
Comment (RaisinTen) [inline shell/common/color_util.cc:38]: While we are at it, we could change this to a better name (hard to tell the difference between color_str and color_string :P)
```suggestion
  std::string converted_color_string;
```

File: shell/common/color_util.cc [hunk 2]
```diff
@@ -35,16 +35,13 @@ namespace electron {
...
+  if (IsHexFormatWithAlpha(color_string)) {
+    std::string temp = color_string;
+    int len = color_string.length() == 5 ? 1 : 2;
+    color_str = temp.erase(1, len) + color_string.substr(1, len);
```
Comment (RaisinTen) [inline shell/common/color_util.cc:42]:
```suggestion
    converted_color_string = temp.erase(1, len) + color_string.substr(1, len);
```

File: shell/common/color_util.cc [hunk 3]
```diff
@@ -35,16 +35,13 @@ namespace electron {
...
+  } else {
+    color_str = color_string;
```
Comment (RaisinTen) [inline shell/common/color_util.cc:44]:
```suggestion
    converted_color_string = color_string;
```
also needs a change few lines below

**Review by RaisinTen — COMMENTED** [review]
Is there a test case where the previous version would parse out a faulty color string? I was wondering if this could also be covered with a test maybe here - https://github.com/electron/electron/blob/c9fd255093fc9d9204db8d50b8795a696f76f611/spec-main/api-browser-window-spec.ts#L1079?

**Comment by codebytere** [comment]
@RaisinTen no - the issue is we don't get the alpha value back out so we can't properly compare it. That's something we could eventually consider changing, but probably out of scope of this PR. I've applied the other fix though!

**Pull request merged**